### PR TITLE
extracting logic from activate into separate functions

### PR DESCRIPTION
**Pull request by lhein**
Starting to make code more readable.

Signed-off-by: Lars Heinemann <lhein.smx@gmail.com>

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -560,5 +513,72 @@ function startKubeProxy(): Promise<string> {
 				reject(new Error(data.toString()));
 			});
 		}
-	}); 
+	});
 }
+
+function determinePlatform(): void {
```
Comment (apupier) [inline src/extension.ts]: The real goal is more to determine the command curl based on the detected platform than to determine platform.

File: src/extension.ts [hunk 2]
```diff
@@ -560,5 +513,72 @@ function startKubeProxy(): Promise<string> {
...
+	const osType = platform();
+	if (osType === 'win32') {
+		curlCommand = 'cmd';
+		curlOption = '/c';
+	}
+}
+
+function applyStatusBarSettings(): void {
+	// process the workspace setting indicating whether we should use the proxy or CLI
```
Comment (apupier) [inline src/extension.ts]: the comment is unrelated to the code

File: src/extension.ts (hunk 1 above)
Comment (lhein) [reply src/extension.ts]: true. I will rename it.

File: src/extension.ts (hunk 2 above)
Comment (lhein) [reply src/extension.ts]: that is nothing I added. I am just extracting to functions.

Comment (apupier) [reply src/extension.ts]: I would prefer to remove the 3 comment lines that are even more useless as the refactoring is making the code clearer, and allows to easily spot that one of them is incorrect

Comment (lhein) [reply src/extension.ts]: done

**Comment by lhein** [comment]
rebased the PR

File: src/extension.ts [hunk 3]
```diff
@@ -589,4 +542,70 @@ function startListeningForServerChanges() {
...
+}
+
+function applyUserSettings(): void {
+	applyStatusBarSettings();
+	applyProxySettings();
+	applyProxyPortSettings();
+}
+
+function createIntegrationsView(): void {
+	// create the integrations view
```
Comment (apupier) [inline src/extension.ts]: this comment is really really redundant

Comment (lhein) [reply src/extension.ts]: removed

**Pull request merged**